node_modules/oae-zencoder/lib/util.js

```diff
@@ -0,0 +1,121 @@
+/*!
+ * Copyright 2014 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var util = require('util');
+var Zencoder = require('zencoder');
+
+var log = require('oae-logger').logger('oae-zencoder');
+var PreviewConstants = require('oae-preview-processor/lib/constants');
+var TenantsAPI = require('oae-tenants/lib/api');
+var ContentConfig = require('oae-config').config('oae-content');
+
+var ZencoderAPI = require('./api');
+var ZencoderConfig = require('oae-config').config('oae-zencoder');
+
+/**
+ * Get the Zencoder API values that have been configured in the Admin UI.
+ *
+ * @return {Object} The Zencoder configuration for the tenant or the generic config for the host
+ */
+var getConfig = module.exports.getConfig = function(tenantAlias) {
+    var config = ZencoderAPI.getConfiguration();
+
+    if (ZencoderConfig.getValue(tenantAlias, 'zencoder', 'enabled') && ZencoderConfig.getValue(tenantAlias, 'zencoder', 'apiKey')) {
+        return _.extend(config, {
+            'apiKey': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'apiKey'),
+            'watermark': null,
+            'amazons3-region': ContentConfig.getValue(tenantAlias, 'storage', 'amazons3-region') || config['amazons3-region'],
+            'amazons3-bucket': ContentConfig.getValue(tenantAlias, 'storage', 'amazons3-bucket') || config['amazons3-bucket'],
+            'amazons3-access-key': ContentConfig.getValue(tenantAlias, 'storage', 'amazons3-access-key') || config['amazons3-access-key'],
+            'amazons3-secret-key': ContentConfig.getValue(tenantAlias, 'storage', 'amazons3-secret-key') || config['amazons3-secret-key']
```

Comment:
Tenants didn't get MediaCore by default. For any tenant that didn't have MediaCore, it wouldn't attempt to generate a preview for video files.

In this particular case, the intention for *Unity is to give all tenants a default ZenCoder API configuration (with limited use) and allow them to override this with their tenant specific ZenCoder configuration. However, the ESUP deployment will not have a default ZenCoder API configuration and will require individual tenants to set this.

Therefore, I think we just want to have a ZenCoder config that can be set at the global admin level and can be overridden at the tenant level.